app/src/main/java/com/example/dogadoption/repository/DogRepository.kt

```diff
@@ -0,0 +1,32 @@
+package com.example.dogadoption.repository
+
+import android.app.Application
+import com.example.dogadoption.retrofit.DogApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class DogRepository @Inject constructor(
+    private val dogApi: DogApi,
+    private val appContext: Application
+) {
+    suspend fun getDogBreeds(): List<String> {
+        return dogApi.getDogBreeds().message.keys.toList()
+    }
+
+    suspend fun getDogPictures(breed: String): List<String> {
+        return withContext(Dispatchers.IO) {
```

Comment:
Functions in the `DogRepository` are already `suspend` so you can leave responsibility of setting the `Dispatcher` to the caller, which is a ViewModel. Or if you really want to you should pass the dispatcher in the `DogRepository` constructor so you can use different dispatcher somewhere else. for example later in tests.

+1 for using `Dispatchers.IO` but why only in `getDogPictures`?